python/test/unit/fem/test_interpolation.py

```diff
@@ -0,0 +1,41 @@
+import cffi
+from petsc4py import PETSc
+import numpy as np
+
+import ufl
+import dolfin
+
+
+def test_interpolation_laplace():
+    mesh = dolfin.generation.UnitSquareMesh(dolfin.MPI.comm_world, 5, 5)
+
+    P2 = dolfin.FunctionSpace(mesh, ("P", 2))
+    dP0 = dolfin.FunctionSpace(mesh, ("DP", 0))
+
+    u = dolfin.Function(P2)
+    f = dolfin.Function(dP0)
+
+    def evaluate(x):
+        return x[:, 0] ** 2
+
+    u.interpolate(evaluate)
+    ufl_expr = ufl.div(ufl.grad(u))
+
+    # Attach the expression with cell midpoint
+    expr = (ufl_expr, np.array([[0.25, 0.25]]))
+
+    module = dolfin.jit.ffc_jit(expr)
+    kernel = module.tabulate_expression
+
+    ffi = cffi.FFI()
+
+    L = dolfin.cpp.fem.Form([dP0._cpp_object])
+    L.set_tabulate_cell(-1, ffi.cast("uintptr_t", kernel))
+    L.set_coefficient(0, u._cpp_object)
+
+    dolfin.fem.assemble_vector(f.vector, L, mode=dolfin.cpp.fem.InsertMode.set)
+
+    # Update ghosts with the same mode as assembly
```

Comment:
Hmm, what do you mean with the mode "update ghosts"? The standard assembler with "add" mode needs update ghosts with add mode, "insert" assembler needs insert ghost update.

Ghost update with add wouldn't work in this case, as for interpolation local information must be forced to the global vector, not added. I am missing something?

Reply:
`ghostUpdate(addv=PETSc.InsertMode.INSERT, mode=PETSc.ScatterMode.FORWARD)`  means  that you're updating the ghost values with the values from the owning processes. It's just the standard mode to update the values of the ghosts. And It would not make sense to use add in that case. 

Maybe you're referring to  `ScatterMode.REVERSE`, in the sense of applying the ghost values to the owning process.

Reply:
Ah, correct, I should be scattering with REVERSE mode. Insert local ghost information back into the global vector. Thats what you mean?

Reply:
I'm also not clear why you even need to update ghosts for `dP0` space?

Reply:
Not needed to communicate... :) I'll remove.

Reply:
And also, applying reverse scatter mode with `INSERT`, in a ghosted mesh, would make the tests to fail.